[Bug] Add BestLocalAddress when responding to getAddr messages

zquestz opened the pull request:
Currently, we don't send our own IP when receiving getAddr messages, this makes sure to include it in the address list.

Testing this now. =)

EDIT: And I have been seen!

```
"35.202.172.160:8333":{"version":"\/bchd:0.13.0\/","country":"US","height":557436,"flags":293}
```

emergent-reasons commented on server.go:
```diff
@@ -1262,6 +1262,17 @@ func (sp *serverPeer) OnGetAddr(_ *peer.Peer, msg *wire.MsgGetAddr) {
+	if bestAddress.Port != 0 {
+		addrCache = addrCache[1:]
+		addrCache = append(addrCache, bestAddress)
+	}
```
I think it doesn't really matter, but this will unnecessarily trim 1 if we aren't already over the limit in `AddressCache`, right?

zquestz replied:
Yep, but I didn't want to export the const for the max addr cache size. Figured it doesn't really matter as there are plenty of addresses already being sent.

zquestz commented:
Ok, I changed my mind. Would love some commentary. I reverted my node to master and it immediately disappeared again. No bchd nodes were found. I redeployed this branch, and the node starting showing up within 2 minutes. I think this should probably go in, even if we get crawled a lot more. Any comments? @cpacia @tyler-smith

nshopik commented:
Will it report correct address if there many of them? what about IPv6 only addresses how does it select which one to report?

zquestz commented:
@nshopik the GetBestLocalAddress handles this for us. It handles IPv4/IPv6 and checks routeability!
